SWG-Source/dsrc
# GU 16.9 - No Trade Removable

**Summary**: Provides a mechanism for the no trade flag to be removed from items that were previously no trade at creation.

![image](https://user-images.githubusercontent.com/762130/93809567-5491b680-fc45-11ea-8ad6-6bbfb6df438a.png)


In the final few years of Galaxies, there were a large number of items that had their no trade status removed. However, these changes were not retroactive on existing items (to the best of my knowledge). So a separate system was devised that works in a similar manner to the No Trade Shared mechanic. If the user was in possession of a no trade item, then it could be radialed to have the no trade status permanently removed.

This change will have little impact on many servers using SWG Source, as the items affected by this change may well have been tradeable from the time of the server's inception. However, the mechanism for allowing operators to specify items that can have their no trade status removed is still useful.

Much of this PR is me looking to immerse myself a little bit deeper into the SWG codebase and understand things better - please feel free to point out any foibles or major mistakes!

Some notes:

- This should not change the behaviour of objects before they've had their no trade status removed.
- Only an item's owner can remove the no trade status.
    - This applies even if the item is no trade shared(TODO: Correct behaviour?)
- This PR set utilizes SERVER_MENU47 to avoid needing synchronized changes across multiple repos.

This change comprises a few different components:

- Changes to `dsrc` to add a script that handles object menu requests for no trade removable items and adds a datatable containing a list of all no-trade removable items.
- Changes to `src` that loads the no trade removable table and then attaches the script to any relevant objects. Also handles sending the no_trade_removable attribute for the examine/details pane
- Small modification to `client-assets` and `serverdata` to add a missing string.
- Changes to `client-tools` to display the No Trade Removable badge within the examine/details pane.

[Table of Items Affected](https://www.notion.so/9d875faaa75b46aca8beb84e580ca42d)

## Testing instructions

1. Spawn an item from within the `no_trade_removable` table. e.g. `/qatool spawnItem item_pgc_planet_hologram_corellia`
2. Make it no trade by using: `/objvar set <oid> noTrade 1`
3. You should be able to return it to it's tradable state by radialing it and selecting the appropriate option within your inventory.
